### PR TITLE
[powerdns] Introduce Powerdns chart

### PR DESCRIPTION
**Pull request by carpenike**
#### Special notes for your reviewer:
Chart includes either mysql (default) or postgres support. May potentially add powerdns-admin in the future.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[radarr]`)

### CODE REVIEW

**Comment by carpenike** [comment]
@billimek -- not sure how to fix this one; is there some regex on the maintainers field?

File: charts/powerdns/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+appVersion: "latest"
+description: PowerDNS is a DNS server, written in C++ and licensed under the GPL. It runs on most Unix derivatives. PowerDNS features a large number of different backends ranging from simple BIND style zonefiles to relational databases and load balancing/failover algorithms. A DNS recursor is provided as a separate program.
+name: powerdns
+version: 0.3.0
+home: https://www.powerdns.com/
+sources:
+  - http://www.github.com/PowerDNS/
+icon: https://www.powerdns.com/img/pDNS_logo_positive_claim_down_screen_xsmall.png
+maintainers:
+  - name: Ryan Holt
```
Comment (billimek) [inline charts/powerdns/Chart.yaml:11]: From what I recall, this needs to be a github username, so `carpenike`

Comment (carpenike) [reply charts/powerdns/Chart.yaml]: aha, ok. trying again. ty.

**Comment by carpenike** [comment]
@billimek -- does the dind not work with dependent charts?

**Comment by billimek** [comment]
@carpenike I think I fixed the dependent chart failure with https://github.com/billimek/billimek-charts/commit/4e667d9f0ef872b80660050acc5c309100c4f249.  The helm linting should be clean now!

File: charts/powerdns/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              value: {{if (.Values.powerdns.dnssec)}}"yes"{{else}}"no"{{ end }}
+{{ end }}
+          ports:
+            - name: dns-tcp
+              containerPort: 53
+              protocol: TCP
+            - name: dns-udp
+              containerPort: 53
+              protocol: UDP
+{{ if (.Values.powerdns.api_key) }}
```
Comment (billimek) [inline charts/powerdns/templates/deployment.yaml:98]: It appears that if the `powerdns.api_key` value is not set, we will never have a `dns-webserver` port defined and thus the probes won't properly work.

If this is required, we may not be able to do automated testing of the chart unless we have some default value for `powerdns.api_key` or work probes in a way that don't require this port?

What do you think?

Comment (carpenike) [reply charts/powerdns/templates/deployment.yaml:98]: Hmm. Theoretically you don't need the API to run powerdns, but I think we should be opinionated here and always run it. What if I auto generate the key if it's not provided?

Thoughts?

**Comment by carpenike** [comment]
@billimek -- Ok, so I think I know what the problem is...

The configmap https://github.com/billimek/billimek-charts/pull/308/files#diff-189124285fdbc305652e7e00a84a2b9c contains the settings which enable the webserver. However, it appears that powerdns will not apply those settings until after it loads for the first time and recognizes the `include: /etc/pdns/conf.d` parameter, which requires a reboot.

On my local nodes that's fine as the timeout will eventually complete and the pod will reboot and apply the config as expected. In the case of the automated tests it appears that this causes the test to fail as the pod never restarts?

Thouhgts?

**Comment by billimek** [comment]
> Thouhgts?

It's super frustrating because it all installs cleanly with the default values when manually running a `helm install` on my cluster.  We may need to just merge this with the 'failing' test.  It's also possible to specify a separate values for CI purposes only (e.g. [frigate](https://github.com/billimek/billimek-charts/blob/master/charts/frigate/ci/ci-values.yaml)).  

Is there some combination of values that that makes this happy for CI?  Otherwise I propose we merge as-is.

**Comment by carpenike** [comment]
Yup that might work. I'll set those as default for the probes and see if it fixes it.

Is there some type of variable that gets set by the tests that we could key off of to disable the probes?

**Comment by carpenike** [comment]
Sorry, I should have looked before asking. Appears that the purpose of the ci path is to override values during CI only. I’ll give it a try today!

Will also make the probe values configurable similar to frigate.

**Comment by carpenike** [comment]
@billimek -- this is strange, honestly can't see any errors in the logs for why it failed. I believe I have the probes fully disabled now as well.

**Comment by billimek** [comment]
Ok I should have checked this earlier.  The `kind` cluster can't (yet) support services of type `LoadBalancer`.  

I suggest that we make the service type `ClusterIP`  in the `ci-values.yaml` in order to it to not get stuck trying to get an external LoadBalancerIP forever.

**Comment by carpenike** [comment]
hahaha.... perfect! now it's green. :-D

**Comment by carpenike** [comment]
@billimek -- all checks pass with probes back on. Should be good to go but let me know if you think I should adjust anything else.

Also rev'd to 1.0.0.

**Pull request merged**